### PR TITLE
iam module

### PR DESCRIPTION
**Pull request by juliedavila**
There is currently no IAM module. This is part 1 of 3 that I plan to make.

Part 1: Create/Update/Delete users, groups and roles
Part 2: Create/Update/Delete policy objects for users/groups/roles
Part 3: Upload/Remove/Update server and signing certificates

I opted for compartmentalization to avoid entering ec2_vpc's monolithic state/complexity.

I've done a good amount of testing but def should get poked some more.

### CODE REVIEW

File: cloud/amazon/iam.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+  access_key_state:
+    description:
+      - when type is user: creates, removes, deactivates or activates a user's access key. Note that remove and deactivate will apply to all keys.
+    required: "false"
+    default: null
+    choices: ["create", "remove", "activate", "deactivate"]
+    aliases: []
+  group:
+    description:
+      - when type is user, adds user to the specified group. When 'update' is specified, it will add the user to groups, preserving original group membership.
```
Comment (mpavlov) [inline cloud/amazon/iam.py]: The distinction between `state == present` and `state == update` is a bit unclear. Is it that `state == present` will remove unspecified group memberships for this user?

File: cloud/amazon/iam.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+## Create, or Delete Roles. There is no update as boto does not currently provide a role_update function
+    if iam_type == 'role' and state == 'present':
+        if name not in role_list:
+            result = create_role(iam, name, path)
+    elif iam_type == 'role' and state == 'absent':
+        if name in role_list:
+            result = delete_role(iam, name)
+
+    module.exit_json(**result)
```
Comment (mpavlov) [inline cloud/amazon/iam.py]: To increase readability and maintainability of `main()`, could factor out the `iam_type`-specific code into dedicated functions. E.g., `work_on_role()`, `work_on_group()`, and `work_on_user()`.

File: cloud/amazon/iam.py (hunk 1 above)
Comment (juliedavila) [inline cloud/amazon/iam.py]: Yeah, will tweak that soon.

File: cloud/amazon/iam.py (hunk 2 above)
Comment (juliedavila) [inline cloud/amazon/iam.py]: I'll have to think about this, I don't really see too much complexity here.

**Comment by juliedavila** [comment]
still some work to be done

**Comment by juliedavila** [comment]
Going to resubmit with cleaner commit history. Now at https://github.com/ansible/ansible-modules-core/pull/756

**Pull request closed without merging**